Foundation/NSString.swift

```diff
@@ -1256,7 +1256,33 @@ extension NSString {
     }
     
     public convenience init(contentsOf url: URL, usedEncoding enc: UnsafeMutablePointer<UInt>?) throws {
-        NSUnimplemented()    
+        let readResult = try NSData(contentsOf: url, options:[])
+
+        let bytePtr = readResult.bytes.bindMemory(to: UInt8.self, capacity:readResult.length)
+        if bytePtr[0] == 254 && bytePtr[1] == 255 {
+          enc?.pointee = String.Encoding.utf16BigEndian.rawValue
+        }
+        else if bytePtr[0] == 255 && bytePtr[1] == 254 {
+          enc?.pointee = String.Encoding.utf16LittleEndian.rawValue
+        }
+        else {
+          //Need to work on more conditions. This should be the default
+          enc?.pointee = String.Encoding.utf8.rawValue
+        }
+
+        guard let cf = CFStringCreateWithBytes(kCFAllocatorDefault, bytePtr, readResult.length, CFStringConvertNSStringEncodingToEncoding(enc!.pointee), true) else {
```

Comment:
This has a guard here, but uses enc! in the dereferencing of the pointer. It would make sense to have let enc = enc, cf = ... enc, so that if enc isn't passed then we don't blow up the process.